### PR TITLE
Added support for Java Flight Recorder features

### PR DESCRIPTION
**Pull request by pivotal-david-osullivan**
## Summary

This PR adds support for Java Flight Recorder features by adding flag `-XX:StartFlightRecording=`.

Support for configuring JFR via arguments is provided with optional environment variable `BPL_JFR_ARGS`. This accepts a comma separated list of any/all of the following arguments and their values, e.g. `filename=/tmp/rec.jfr,dumponexit=true`:

- disk
- dumponexit
- filename
- maxsize
- maxage
- settings
- path-to-gc-roots
- name
- delay

## Use Cases
Enable Java Flight Recording for analysis via JMX or jcmd tool

## Checklist

* [ ] I have viewed, signed, and submitted the Contributor License Agreement.
* [ ] I have linked issue(s) that this PR should close using keywords or the Github UI (See [docs](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue))
* [ ] I have added an integration test, if necessary.
* [ ] I have reviewed the [styleguide](https://github.com/paketo-buildpacks/community/blob/main/STYLEGUIDE.md) for guidance on my code quality.
* [ ] I'm happy with the commit history on this PR (I have rebased/squashed as needed).

### CODE REVIEW

File: helper/jfr.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+	"github.com/paketo-buildpacks/libpak/bard"
+)
+
+type JFR struct {
+	Logger bard.Logger
+}
+
+func (j JFR) Execute() (map[string]string, error) {
+	if val, ok := os.LookupEnv("BPL_JAVA_FLIGHT_RECORDER_ENABLED"); !ok || val != "true" {
```
Comment (dmikusa-pivotal) [inline helper/jfr.go:32]: I know the rfc says to use this env variable but it is pretty long. What do you think about shortening this to `BPL_JFR_ENABLED`?

It's shorter, but it is less discoverable, so I'm not 100% sold on it.

File: helper/jfr.go [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+		return nil, nil
+	}
+
+	argList := sherpa.GetEnvWithDefault("BPL_JFR_ARGS", "")
```
Comment (dmikusa-pivotal) [inline helper/jfr.go:36]: Whatever we end up doing with the `BPL_*_ENABLED` env variable, we need to make sure this one uses a consistent prefix. That might be updating this one to `BPL_JAVA_FLIGHT_RECORDER_ARGS` or it might be updating the former. They should have the same prefix though.

File: helper/jfr.go [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+	j.Logger.Infof("Enabling Java Flight Recorder")
+
+	// minimum flag to enable JFR, with default config args
+	jfrConfig := "-XX:StartFlightRecording="
```
Comment (dmikusa-pivotal) [inline helper/jfr.go:41]: If you do not specify any `BPL_JFR_ARGS`, the default behavior seems to require a user to connect with `jcmd` or remotely with JMC to get a recording. 

I'm trying to think of some default we could use that might be easier to consume without further interaction 🤔 

I had a few thoughts, but the simplest one would seem to be just dumping on exit. We can enable the `dumponexit=true` option. This would trigger a flight recording to be dumped on exit to the `/workspace` directory. We would probably also need to set `filename=/tmp/` so that it writes to the temp directory which would be writable for a non-cnb user, although perhaps not? Perhaps that's up to the user if the user chooses to not be the cnb user.

   At any rate, this would result in JFR running and dumping on exit if there is a problem, similar to what we do with Java NMT. I don't believe this writes anything to disk, except on exit, so that keeps disk usage and I/O low.

tl;dr if `BPL_JFR_ARGS` is empty, then we should set `dumponexit=true` to the default. I think it would be sufficient to either get default or fully custom (i.e. don't try merging user settings and the defaults).

**Review by dmikusa-pivotal — CHANGES_REQUESTED** [review]
I think we're on the right path. A few comments for things I think we can improve. 

I also got some test errors:

```
> richgo test ./...
PASS | github.com/paketo-buildpacks/libjvm
PASS | github.com/paketo-buildpacks/libjvm/calc
SKIP | github.com/paketo-buildpacks/libjvm/cmd/helper	[no test files]
PASS | github.com/paketo-buildpacks/libjvm/count
     | Suite: libjvm/helper
     | Total: 70 | Focused: 0 | Pending: 0
     | ...................................................................xx.
     | Passed: 68 | Failed: 2 | Skipped: 0
FAIL | Unit (0.04s)
FAIL |     Unit/libjvm/helper (0.03s)
FAIL |             Unit/libjvm/helper/JFR/$BPL_JAVA_FLIGHT_RECORDER_ENABLED/$BPL_JFR_ARGS_is_set/returns_an_error_if_a_JFR_argument_is_empty (0.00s)
     |             jfr_test.go:62:
     |                 Expected an error to have occurred.  Got:
     |                     <nil>: nil
FAIL |             Unit/libjvm/helper/JFR/$BPL_JAVA_FLIGHT_RECORDER_ENABLED/$BPL_JFR_ARGS_is_set/returns_an_error_if_JFR_arguments_are_not_parsable (0.00s)
     |             jfr_test.go:68:
     |                 Expected an error to have occurred.  Got:
     |                     <nil>: nil
FAIL | 	github.com/paketo-buildpacks/libjvm/helper	0.242s
SKIP | github.com/paketo-buildpacks/libjvm/internal	[no test files]
```

I think CI is failing similarly.

File: helper/jfr.go (hunk 2 above)
Comment (pivotal-david-osullivan) [reply helper/jfr.go:36]: Going with shorter version `BPL_JFR_*` for these

**Review by dmikusa-pivotal — APPROVED** [review]
